Reachable unwrap in `FsResolver::create_new_file()`

### Describe the bug
There is a reachable panic in `FsResolver::create_new_file()` at kernel/src/fs/fs_resolver.rs:134 when call syscall `openat`.

https://github.com/asterinas/asterinas/blob/ae4ac384713e63232b74915593ebdef680049d31/kernel/src/fs/fs_resolver.rs#L134

This is introduced in commit 2c6dd07.
### To Reproduce
1. Compile a program which calls `openat`: 
```C
#define _GNU_SOURCE
#include <errno.h>
#include <fcntl.h>
#include <stdio.h>
#include <sys/syscall.h>
#include <unistd.h>

int main() {
  FILE *fp = fopen("/proc/sys", "r");
  int dfd = fileno(fp);
  fcntl(dfd, F_SETFL, 0x44c00);
  const char *filename = ".//dev/random";
  int flags = 0x81040;
  mode_t mode = 0666;
  int fd = syscall(SYS_openat, dfd, filename, flags, mode);

  perror("openat");
  return 0;
}
```
2. Run the compiled program in Asterinas.

### Expected behavior

Asterinas reports panic and is terminated.

### Environment
- Official docker asterinas/asterinas:0.8.3
- 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
- Asterinas version: main ae4ac38

### Logs
```
panicked at /root/asterinas/kernel/src/fs/fs_resolver.rs:134:42:
called `Option::unwrap()` on a `None` value
Printing stack trace:
   1: fn 0xffffffff887859c0 - pc 0xffffffff88785a2a / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017a9310;

   2: fn 0xffffffff887857d0 - pc 0xffffffff88785920 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017a9370;

   3: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017a94c0;

   4: fn 0xffffffff889a20f0 - pc 0xffffffff889a2172 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017a94d0;

   5: fn 0xffffffff889a2290 - pc 0xffffffff889a22d0 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017a9560;

   6: fn 0xffffffff8898bae0 - pc 0xffffffff8898baf5 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017a95d0;

   7: fn 0xffffffff885e4cb0 - pc 0xffffffff885e4d8e / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017a95e0;

   8: fn 0xffffffff885e4080 - pc 0xffffffff885e4491 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017a97e0;

   9: fn 0xffffffff8811fe70 - pc 0xffffffff881206c0 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017a9a50;

  10: fn 0xffffffff88063b80 - pc 0xffffffff88091cef / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017a9f30;

  11: fn 0xffffffff88061d50 - pc 0xffffffff88061dde / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017c03d0;

  12: fn 0xffffffff88402130 - pc 0xffffffff88402cf6 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017c0570;

  13: fn 0xffffffff88476190 - pc 0xffffffff8847619e / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017c0f90;

  14: fn 0xffffffff887ad5c0 - pc 0xffffffff887ad5d6 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017c0fb0;

  15: fn 0xffffffff887ef6e0 - pc 0xffffffff887ef749 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017c0fd0;


     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff88864a27; rbp                0x0; rsp 0xffff8000017c1000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panicking.rs:113
(  2) /root/asterinas/ostd/src/panicking.rs:59
(  3) ezxtz4txsqany8s5wl6db2xrn:?
(  4) ??:?
(  5) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:220
(  6) ??:?
(  7) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:960
(  8) /root/asterinas/kernel/src/fs/fs_resolver.rs:69
(  9) /root/asterinas/kernel/src/syscall/open.rs:34
( 10) /root/asterinas/kernel/src/syscall/mod.rs:178
( 11) /root/asterinas/kernel/src/syscall/mod.rs:328
( 12) /root/asterinas/kernel/src/thread/task.rs:69
( 13) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:79
( 14) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2077
( 15) /root/asterinas/ostd/src/task/mod.rs:192
make: *** [Makefile:173: run] Error 1
```

--- a/kernel/src/fs/fs_resolver.rs
+++ b/kernel/src/fs/fs_resolver.rs
@@ -131,7 +131,9 @@ impl FsResolver {
             return_errno_with_message!(Errno::EISDIR, "path refers to a directory");
         }
 
-        let parent = lookup_ctx.parent().unwrap();
+        let parent = lookup_ctx
+            .parent()
+            .ok_or_else(|| Error::with_message(Errno::ENOENT, "parent not found"))?;
         if !parent.mode()?.is_writable() {
             return_errno_with_message!(Errno::EACCES, "file cannot be created");
         }

